app/graphql/types/invoices/object.rb

```diff
@@ -83,6 +83,29 @@ def payable_type
         "Invoice"
       end
 
+      # rubocop:disable GraphQL/ResolverMethodLength
+      def customer
+        object.customer.tap do |c|
+          c.display_name = object.customer_display_name
+          c.firstname = object.customer_firstname
+          c.lastname = object.customer_lastname
+          c.email = object.customer_email
+          c.phone = object.customer_phone
+          c.url = object.customer_url
+          c.tax_identification_number = object.customer_tax_identification_number
+          c.timezone = object.customer_applicable_timezone
+          c.address_line1 = object.customer_address_line1
+          c.address_line2 = object.customer_address_line2
+          c.city = object.customer_city
+          c.state = object.customer_state
+          c.zipcode = object.customer_zipcode
+          c.country = object.customer_country
+          c.legal_name = object.customer_legal_name
+          c.legal_number = object.customer_legal_number
+        end
+      end
```

Comment:
What about 

```suggestion
      class CustomerDecorator
        [
          :display_name,
          :firstname,
          # ..
          :legal_number
        ].each { |attr| define_method(attr) { public_send("customer_#{attr}") }
      end
      def customer
        CustomerDecorator.new(object.customer)
      end
```

?

That way we don't need a trick on the `Invoice` model to please GraphQL 😉

Reply:
@julienbourdeau I forgot to add `< SimpleDelegator` in my suggestion 🤦 

```ruby
      class CustomerDecorator < SimpleDelegator
        [
          :display_name,
          :firstname,
          # ..
          :legal_number
        ].each { |attr| define_method(attr) { public_send("customer_#{attr}") }
      end
      def customer
        CustomerDecorator.new(object.customer)
      end
```

So no issue with other methods.

Reply:
sure thing, I have added a delegator with simple delegator and remove the attribute writer for display_name in customer... I did not like it tbh.